src/tokenized-line.coffee

```diff
@@ -97,7 +98,7 @@ class TokenizedLine
         tokenOffset = 0
 
       # split out leading soft tabs
-      else if charCode is SpaceCharCode
+      else if charCode is SpaceCharCode or charCode is NonBreakingSpaceCharCode
```

Comment:
Can you use two different cases ?

issue  #8080 basically state that non breaking space are undesirable as they break code
so maybe we should use a different visual than plain space.